libbeat/autodiscover/providers/kubernetes/kubernetes.go

```diff
@@ -165,18 +165,31 @@ func (p *Provider) String() string {
 	return "kubernetes"
 }
 
-func (p *Provider) publish(event bus.Event) {
-	// Try to match a config
-	if config := p.templates.GetConfig(event); config != nil {
-		event["config"] = config
-	} else {
-		// If there isn't a default template then attempt to use builders
-		e := p.eventManager.GenerateHints(event)
-		if config := p.builders.GetConfig(e); config != nil {
-			event["config"] = config
+func (p *Provider) publish(events []bus.Event) {
+	if len(events) == 0 {
+		return
+	}
+
+	configs := make([]*common.Config, 0)
+	for _, event := range events {
+		// Try to match a config
+		if config := p.templates.GetConfig(event); config != nil {
+			configs = append(configs, config...)
+		} else {
+			// If there isn't a default template then attempt to use builders
+			e := p.eventManager.GenerateHints(event)
+			if config := p.builders.GetConfig(e); config != nil {
+				configs = append(configs, config...)
+			}
 		}
 	}
 
+	// Since all the events belong to the same event ID pick on and add in all the configs
+	event := bus.Event(common.MapStr(events[0]).Clone())
+	// Remove the port to avoid ambiguity during debugging
+	delete(event, "port")
+	event["config"] = configs
```

Comment:
Wondering if we could avoid this merge and check the configs differently in autodiscover runner. Feel like this will make things more complex and hard for debugging in the future. wdyt?

Reply:
the problem is that we wouldnt know if a config was removed during a resync if we dont compare them together.